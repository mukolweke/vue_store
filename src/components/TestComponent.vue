<template>
    <div>
        <span>{{ message }}</span>

        <button @click="clickMe">Click Me</button>
        <ul>
            <li style="margin-top: 10px" class="message" @click="handleClick">
                {{info}}
            </li>
        </ul>


    </div>

</template>

<script>
    export default {
        name: 'TestComponent',
        props: [{
            info: {
                type: String,
                required: true,
                validator: value => value.length > 1
            },
            info2: String,

            movie: {
                type: String,
                default: 'Avengers'
            }
        }],
        data() {
            return {
                message: '',
                info: ''
            }
        },
        created() {
            this.message = 'Vesper'
        },
        methods: {
            clickMe: () => {
                return "kuku"
            },
            handleClick() {
                console.log('lalala')
                this.$emit('message-clicked', this.info)
            }
        }
    }
</script>