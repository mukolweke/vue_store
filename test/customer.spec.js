import Vue from 'vue'
import {shallow} from 'vue-test-utils'
import sinon from 'sinon'
import TestComponent from '../src/components/TestComponent'

describe('TestComponent', () => {
    let ClonedComponent, vm

    beforeEach(() => {
        ClonedComponent = shallow(TestComponent,{
            data: {
                message: 'Vesper'
            }
        })
    })

    // name of the component
    it('is called TestComponent', () => {
        const wrapper = shallow(TestComponent);

        expect(wrapper.name()).toBe('TestComponent')
    })

    // test if button has been clicked
    it('should change the message when clicked', function () {

        const listener = sinon.spy();

        const  wrapper = shallow(TestComponent)

        wrapper.setMethods({
            $emit: listener,
        })

        // click button
        wrapper.find('button').trigger('click')

        expect(listener.calledWith('addme'));

    });

    // test if you can change the messages attribute in data
    it('equals message to ["Vesper"]', () => {

        expect(ClonedComponent.vm.message).toEqual('Vesper')


    })

    // check the output in snapshot
    it('has the expected html structure', () => {
        expect(ClonedComponent.element).toMatchSnapshot()
    })



    it('has a created hook', () => {

        expect(typeof TestComponent.methods.clickMe).toBe('function')

    })


})