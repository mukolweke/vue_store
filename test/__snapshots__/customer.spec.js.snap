// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestComponent has the expected html structure 1`] = `
<div>
  <span>
    Vesper
  </span>
   
  <button>
    Click Me
  </button>
</div>
`;
